### PR TITLE
CDAP-12360 Fix logging deadlock in CDAP master.

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-12360
I was able to reproduce the failure and validated that these changes fix the issue.

https://builds.cask.co/browse/CDAP-RUT1260-2

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogMessage.java [hunk 1]
```diff
@@ -118,6 +119,13 @@ public long getTimeStamp() {
   @Override
   public void prepareForDeferredProcessing() {
     loggingEvent.prepareForDeferredProcessing();
+    Object[] args = loggingEvent.getArgumentArray();
+    if (args != null && argumentArray != null) {
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogMessage.java]: should be `argumentArray == null`.

**Comment by chtyim** [comment]
Just one comment.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogMessage.java (hunk 1 above)
Comment (anwar6953) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/LogMessage.java]: oops. Flipped the condition now.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**